Source/messaging/ConsoleStreamRedirect.h

```diff
@@ -0,0 +1,498 @@
+	if (_index != Core::IResource::INVALID) {
+		::fsync(_copy);
+		::dup2(_copy, _index);
+		::close(_copy);
+		Core::ResourceMonitor::Instance().Unregister(*this);
+		_index = Core::IResource::INVALID;
+		Reader::Flush();
+	}
+	return (_index == -1);
+}
+
+Core::IResource::handle Descriptor() const override {
+	return (_handle);
+}
+uint16_t Events() override {
+	if (_handle != Core::IResource::INVALID) {
+		return (POLLHUP | POLLRDHUP | POLLIN);
+	}
+	return (0);
+}
+void Handle(const uint16_t events) override {
+	// If we have an event, read and see if we have a full line..
+	if ((events & POLLIN) != 0) {
+		int readBytes, freeSpace;
+
+		do {
+			readBytes = read(_handle, Reader::Buffer(), Reader::Length());
+
+			if (readBytes > 0) {
+				Reader::ProcessBuffer(readBytes);
+			}
+
+		} while (readBytes == freeSpace);
+	}
+}
+
+			private:
+				Core::IResource::handle _index;
+				Core::IResource::handle _copy;
+				Core::IResource::handle _handle;
+				bool _register;
+			};
+#endif
+
+		public:
+			ConsoleStreamRedirectType(ConsoleStreamRedirectType<STREAMTYPE>&&) = delete;
+			ConsoleStreamRedirectType(const ConsoleStreamRedirectType<STREAMTYPE>&) = delete;
+			ConsoleStreamRedirectType<STREAMTYPE> operator=(ConsoleStreamRedirectType<STREAMTYPE>&&) = delete;
+			ConsoleStreamRedirectType<STREAMTYPE> operator=(const ConsoleStreamRedirectType<STREAMTYPE>&) = delete;
+
+			ConsoleStreamRedirectType() 
+				: _channel(*this) {
+			}
+			~ConsoleStreamRedirectType() = default;
+
+		public:
+			bool Open(const Core::IResource::handle fileDescriptor) {
+				return (_channel.Open(fileDescriptor));
+			}
+			bool Close() {
+				return (_channel.Close());
+			}
+
+		private:
+			void Output(const uint16_t length, const TCHAR buffer[]) {
+				_metadata.Output(length, buffer);
+			}
+
+		private:
+			ReaderImplementation _channel;
+			STREAMTYPE _metadata;
+		};
+
+		class EXTERNAL StandardOut {
+		public:
+			StandardOut(StandardOut&&);
+			StandardOut(const StandardOut&);
+			StandardOut& operator=(StandardOut&&);
+			StandardOut& operator=(const StandardOut&);
+
+			StandardOut() = default;
+			~StandardOut() = default;
+
+		public:
+			void Output(const uint16_t length, const TCHAR buffer[]) {
+				// if (LocalLifetimeType<STREAMTYPE, &Core::System::MODULE_NAME, Core::Messaging::Metadata::type::TRACING>::IsEnabled() == true) {
+
+				// }
+				string text(buffer, length);
+				fprintf(stderr, "Redirected: \"%s\"\n", text.c_str());
+			}
+
+		};
+
+		class EXTERNAL StandardError {
+		public:
+			StandardError(StandardError&&);
+			StandardError(const StandardError&);
+			StandardError& operator=(StandardError&&);
+			StandardError& operator=(const StandardError&);
+
+			StandardError() = default;
+			~StandardError() = default;
+
+		public:
+			void Output(const uint16_t length, const TCHAR buffer[]) {
+				// if (LocalLifetimeType<STREAMTYPE, &Core::System::MODULE_NAME, Core::Messaging::Metadata::type::TRACING>::IsEnabled() == true) {
```

Comment:
These both `Output()` methods are going to have something similar to the macros for Tracing, Logging, Warning Reporting inside, and they will end with a `MessageUnit::Push()` method?

Reply:
Indeed, here we are going to "Push" the actual stderr/stdout to the buffers using the new values in the enum you already added. Does that make sense?

Reply:
As long as we are checking if this message type `IsEnabled()`, and then sending a metadata object derived from `MessageInfo` plus the actual message as the parameters of `MessageUnit::Push()`, then it will work just the same as tracing, so it does make sense and looks solid :)

Reply:
Indeed the details should still be filled in, but the framework (this template) should already redirect a filedescriptor (stderr or stdout) to the Output() with every line (ending by /n /n/r or /r/n) and not including the ending chars.